jnigen/lib/src/bindings/common.dart

```diff
@@ -30,15 +30,25 @@ abstract class BindingsGenerator {
 
   static const String jthrowableType = '${jni}JThrowablePtr';
 
+  static const String jniStringType = '${jni}JString';
+  static const String jniStringTypeClass = '${jni}JString$typeClassSuffix';
+
   static const String jniObjectType = '${jni}JObject';
+  static const String jniObjectTypeClass = '${jni}JObject$typeClassSuffix';
 
   static const String jniArrayType = '${jni}JArray';
+  static const String jniArrayTypeClass = '${jni}JArray$typeClassSuffix';
 
   static const String jniCallType = '${jni}JniCallType';
 
-  static const String jniTypeType = '${jni}JType';
+  static const String jniTypeType = '${jni}JObjType';
   static const String typeClassSuffix = 'Type';
-  static const String typeClassPrefix = '_\$';
+  // TODO(HosseinYousefi): this is a temporary fix for the name collision
```

Comment:
Please file a GitHub issue to point to.

Rationale:

* It's easier to write more context there that you don't want to spell out in a single line comment.
* It's easier to have a design discussion there if there's some trade-off to be discussed.